libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -171,7 +171,11 @@ int AP_Logger_Write(lua_State *L) {
     lua_Alloc allocf = lua_getallocf(L, nullptr);
     char *buffer = (char*)allocf(nullptr, nullptr, 0, msg_len);
     if (buffer == nullptr) {
-        return luaL_error(L, "Buffer allocation failed");
+        lua_gc(L, LUA_GCCOLLECT, 0);
```

Comment:
Looking at it here this actually seems more complicated then we needed. Why not use `luaM_new` rather then manually getting the allocator? This will automatically invoke a GC on fail for us, and should be simpler then all these additions.

Reply:
I've added a patch to do that, please check